src/conf_mode/load-balancing-inbound-ipvsadm.py

```diff
@@ -0,0 +1,140 @@
+    
+    if 'rule' in conf:
+        default_values = defaults(base + ['rule'])
+        for rule in conf['rule']:
+            conf['rule'][rule] = dict_merge(default_values, conf['rule'][rule])
+    
+    return conf
+
+def verify(config):
+    if 'rule' not in config:
+        return None
+    
+    ip_type = None
+    virtual_server_list = []
+    vip_list = []
+    for rule, rule_config in config['rule'].items():
+        if 'disable' in rule_config:
+            continue
+            
+        err_msg = f'Error in the configuration of rule {rule}. '
+        if 'virtual_address' not in rule_config or 'port' not in rule_config:
+            raise ConfigError(f'{err_msg} The virtual server IP and port must exist!')
+            
+        virtual_ip = rule_config['virtual_address']
+        virtual_port = rule_config['port']
+        virtual_server = f'{bracketize_ipv6(virtual_ip)}:{virtual_port}'
+        
+        if virtual_server not in virtual_server_list:
+            virtual_server_list.append(virtual_server)
+        else:
+            raise ConfigError(f'{err_msg} Do not allow duplicate virtual servers!')
+        
+        if virtual_ip not in vip_list:
+            vip_list.append(virtual_ip)
+        
+    for rule, rule_config in config['rule'].items():
+        if 'disable' in rule_config:
+            continue
+            
+        err_msg = f'Error in the configuration of rule {rule}.'
+        
+        virtual_ip = rule_config['virtual_address']
+        virtual_port = rule_config['port']
+        
+        backend_server_list = []
+        if 'backend' in rule_config:
+            for backend, backend_config in rule_config['backend'].items():
+                if 'disable' in backend_config:
+                    return None
+                
+                backend_server = f'{bracketize_ipv6(backend)}:{virtual_port}'
+                    
+                if backend_server not in backend_server_list:
+                    backend_server_list.append(backend_server)
+                else:
+                    raise ConfigError(f'{err_msg} Do not allow duplicate backend servers!')
+                
+                if backend in vip_list:
+                    raise ConfigError(f'{err_msg} The back-end server cannot use VIP (virtual server IP)!')
+                
+                if is_ipv4(virtual_ip) and not is_ipv4(backend) or is_ipv6(virtual_ip) and not is_ipv6(backend):
+                    raise ConfigError(f'{err_msg} The IP of the backend server {backend} \
+                        of rule {rule} must be in the same address family as the virtual server IP!')
+            
+        if 'algorithm' not in rule_config and 'mode' in rule_config:
+            raise ConfigError(f'{err_msg} mode and algorithm must be configured!')
+        
+        if dict_search('mode', rule_config) == 'tun' and dict_search('tun_option.type', rule_config) =='gue' and \
```

Comment:
Please place every operand behind `and` on a new line for easier readability.

Reply:
done